monkey/common/configuration/agent_sub_configuration_schemas.py

```diff
@@ -12,12 +14,14 @@
     TCPScanConfiguration,
 )
 
+valid_custom_pba_filename_regex = re.compile(r"^([a-zA-Z0-9\ \._-]*)$")
```

Comment:
What's important to validate here? according to this `.........` is a valid filename. I'd rather we don't allow spaces.

Reply:
On Linux, `........` is a valid file name. On Windows, I'm not sure.

We should have different regexes for Windows and Linux. Here are some resources to help:

https://stackoverflow.com/questions/1976007/what-characters-are-forbidden-in-windows-and-linux-directory-names
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_282
https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file

Reply:
![image](https://user-images.githubusercontent.com/19957806/180997314-a431db0a-186f-4438-a4b1-517019f400f5.png)
https://en.wikipedia.org/wiki/Ext3

Reply:
We don't have to accept all possible filenames. It's best to just accept characters that don't need any escaping so no one needs to worry about how the escaping is handled

Reply:
Keep in mind the file is uploaded into the island (linux files can be uploaded into windows island). And the names get sanitized. Ideally, we shouldn't allow any filenames that require sanitization, because users don't need to be exposed to sanitization algorithm.

Reply:
> It's best to just accept characters that don't need any escaping so no one needs to worry about how the escaping is handled

We don't do anything with the file other than store it. Since we're not using a shell to do so, I can't think if anything we need to escape (it's totally possible there's something I haven't considered). If the user wants to upload a complicated file name and construct a command that needs escaping, that's the user's choice.

> linux files can be uploaded into windows island

Good point, though we could work around that.

>  Ideally, we shouldn't allow any filenames that require sanitization

What needs to be sanitized?

> users don't need to be exposed to sanitization algorithm

Then the allowed character set is, in effect, a sanitization algorithm, and it's virtually impossible not to expose that to the user.

Reply:
> What needs to be sanitized?

Ah, I see. https://tedboy.github.io/flask/generated/werkzeug.secure_filename.html